src/azure-cli-core/azure/cli/core/profiles/_shared.py

```diff
@@ -98,21 +97,43 @@ def __init__(self, default_api_version, profile=None):
 }
 
 
+class _ApiVersions(object):  # pylint: disable=too-few-public-methods
+    def __init__(self, client_type, sdk_profile, post_process):
+        self._client_type = client_type
+        self._sdk_profile = sdk_profile
+        self._post_process = post_process
+        self._operations_groups_value = None
+        self._resolved = False
+
+    def _resolve(self):
+        if self._resolved:
+            return
+
+        self._operations_groups_value = {}
+        for operation_group_name, operation_type in self._client_type.__dict__.items():
+            if not isinstance(operation_type, property):
+                continue
+
+            value_to_save = self._sdk_profile.profile.get(
+                operation_group_name,
+                self._sdk_profile.default_api_version
+            )
+            self._operations_groups_value[operation_group_name] = self._post_process(value_to_save)
+        self._resolved = True
+
+    def __getattr__(self, item):
```

Comment:
Just to cover all the bases, no-one is trying to enumerate the attributes of this class, correct?

Reply:
No, as far as the current usage.